xla/service/gpu/buffer_comparator.h

```diff
@@ -30,8 +30,8 @@ class BufferComparator {
   BufferComparator(const BufferComparator&) = delete;
   BufferComparator(BufferComparator&&) = default;
 
-  BufferComparator(const Shape& shape, const HloModuleConfig& config,
-                   double tolerance = 0.1);
+  BufferComparator(const Shape& shape, double tolerance = 0.1,
```

Comment:
The BufferComparator constructor is also called in `conv_algorithm_picker.cc` and `gemm_fusion_autotuner.cc`. Please modify these files to take into account `HloModuleConfig` is no longer an argument.

Also mark the constructor as `explicit`

Reply:
Indeed, looks like these parts of conv_algorithm_picker and gemm_fusion_autotuner are excluded from build on ROCM.. Now I have fixed them on CUDA machine